lib-find-meal/src/main/java/com/kinandcarta/lib/find/meal/view/FindMealActivity.kt

```diff
@@ -17,52 +27,148 @@ import kotlinx.coroutines.launch
 
 class FindMealActivity : AppCompatActivity() {
     private val mainScope = MainScope()
+    private val sdk = MealsSDK()
+    private val mealsAdapter = MealsAdapter(listOf())
 
     private lateinit var mealsRecyclerView: RecyclerView
     private lateinit var progressBarView: FrameLayout
     private lateinit var swipeRefreshLayout: SwipeRefreshLayout
 
-    private val sdk = MealsSDK()
-    private val mealsAdapter = MealsAdapter(listOf())
+    private lateinit var fusedLocationProviderClient: FusedLocationProviderClient
+    private lateinit var locationCallback: LocationCallback
+    private lateinit var locationRequest: LocationRequest
+    private var locationUpdateState = false
+    private lateinit var lastLocation: Location
 
     override fun onCreate(savedInstanceState: Bundle?) {
         super.onCreate(savedInstanceState)
 
-        title = "Find a meal"
-        setContentView(R.layout.find_meal_activity)
+        setContentView(com.kinandcarta.lib.find.meal.R.layout.find_meal_activity)
+        title = getString(R.string.find_meal_title) // TODO figure out how to reach strings from main res/strings
+
+        fusedLocationProviderClient = LocationServices.getFusedLocationProviderClient(this)
+        updateLastLocation()
 
-        mealsRecyclerView = findViewById(R.id.rvMeals)
-        progressBarView = findViewById(R.id.progressBar)
-        swipeRefreshLayout = findViewById(R.id.swipeContainer)
+        locationCallback = object : LocationCallback() {
+            override fun onLocationResult(p0: LocationResult?) {
+                super.onLocationResult(p0)
+
+                if (p0 != null) {
```

Comment:
If you change `lastLocation` to be nullable then you can probably just do `lastLocation = locationResult?.lastLocation` without the need to check if locationResult is empty (I'd also change the p0 arg name to be locationResult).

Alternatively I think you can do 
```kotlin
val location = locationResult?.lastLocation ?: return
lastLocation = location
```